tests/cli.rs

```diff
@@ -509,16 +509,18 @@ fn request_binary_detection() {
         "#});
 }
 
-#[test]
-fn timeout() {
-    get_command()
-        .arg("--timeout=0.01")
-        .arg("https://httpbin.org/delay/0.05")
-        .assert()
-        .failure()
-        .stderr(predicates::str::contains("operation timed out"));
-}
-
+// TODO: figure out a way to avoid non-deterministic test failure
```

Comment:
Why does this fail? Does it manage to connect before the timeout anyway?

reqwest tests this by [connecting to an IP address on the local network with a port that's unlikely to be used](https://github.com/seanmonstar/reqwest/blob/8d3e27966c80a18e4ae71fd40a564d9cd096708a/tests/timeouts.rs#L76). Not airtight, but maybe that would be a better approach?

Reply:
>Why does this fail? Does it manage to connect before the timeout anyway?

Yes, I think we should update `xh` to use reqwest's `timeout` API instead of `connect_timeout`.

### Timeout behavior in HTTPie
```
$ http --timeout=1 httpbin.org/delay/5

http: error: Request timed out (1.0s).
```

### Timeout behavior in xh with `.connect_timeout()`
```
$ xh httpbin.org/delay/5 --timeout=1

HTTP/1.1 200 OK
access-control-allow-credentials: true
access-control-allow-origin: *
connection: keep-alive
content-length: 338
content-type: application/json
date: Wed, 28 Apr 2021 17:35:37 GMT
server: gunicorn/19.9.0

{
    "args": {},
    "data": "",
    "files": {},
    "form": {},
    "headers": {
        "Accept": "*/*",
        "Accept-Encoding": "gzip, br",
        "Host": "httpbin.org",
        "User-Agent": "xh/0.9.2",
        "X-Amzn-Trace-Id": "Root=1-60899ce4-51fea36a5bff4e311d8e0d71"
    },
    "origin": "86.148.212.173",
    "url": "http://httpbin.org/delay/5"
}
```

### Timeout behavior in xh with `.timeout()`
```
$ xh httpbin.org/delay/5 --timeout=1

Error: error sending request for url (http://httpbin.org/delay/5): operation timed out

Caused by:
    operation timed out
```

It would also be nice to match HTTPie error message i.e include the timeout value

Reply:
Yep. I suggested `.connect_timeout()` but I was mistaken about what HTTPie did, `.timeout()` gives exactly the right behavior as far as I can tell.